src/shims/unix/fs.rs

```diff
@@ -1659,6 +1666,138 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             }
         }
     }
+
+    fn mkstemp(&mut self, template_op: &OpTy<'tcx, Tag>) -> InterpResult<'tcx, i32> {
+        use rand::distributions::{Distribution, Uniform};
+
+        let this = self.eval_context_mut();
+
+        let target_is_windows =
+            this.eval_context_ref().tcx.sess.target.families.contains(&Cow::Borrowed("windows"));
```

Comment:
`target_is_windows` is always `false`.
This function should use `assert_target_os_is_unix`, it can never be run on a Windows target to begin with!